addons/crm_livechat/models/crm_lead.py

```diff
@@ -0,0 +1,28 @@
+from ast import literal_eval
+
+from odoo import api, fields, models
+
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    channel_id = fields.Many2one(
+        "discuss.channel",
+        "Live chat from which the lead was created",
+        readonly=True,
+        groups="base.group_erp_manager",
+        index="btree_not_null",
+    )
+    from_livechat = fields.Boolean(compute="_compute_from_livechat", compute_sudo=True)
+
+    @api.depends("channel_id")
+    def _compute_from_livechat(self):
+        for lead in self:
+            lead.from_livechat = bool(lead.channel_id)
+
+    def action_redirect_to_livechat_channel(self):
+        action = self.env["ir.actions.actions"]._for_xml_id("mail.action_discuss")
+        action["context"] = literal_eval(action.get("context", "{}"))
+        # sudo - crm.lead: can read origin channel of the lead
+        action["context"]["active_id"] = self.sudo().channel_id.id
+        return action
```

Comment:
Wouldn't it be functionally better to open the chat window, to keep the lead page visible at the same time?

Reply:
done :+1: